doc/sphinx/source/community/backward_compatibility.rst

```diff
@@ -0,0 +1,263 @@
+related to *backwards-incompatible changes* when *backwards-incompatible
+changes* are introduced to the main branch / when a *release* of
+ESMValTool is created.
+
+Guidance on handling *backward-incompatible changes*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As well as helping users to handle *backward-incompatible changes*, the
+policy and surrounding tools must help developers avoid making
+*backwards-incompatible changes*. Not many ideas developed on this yet,
+but components should include:
+
+-  Testing; *backwards-incompatible changes* should be discovered as
+   early in the development process as possible. This motivates
+   continued investment in automated testing.
+-  Guidance on how to minimise the likelihood of introducing
+   *backward-incompatible changes* and how to use deprecation warnings
+   when needed (see `developer
+   guidance <https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility>`__).
+-  Instructions on how to provide the information that *recipe
+   developers* need to adapt to the *backward-incompatible change*,
+   e.g. the text that will go into *release* notes if we follow the
+   proposal above.
+-  General instructions for *recipe developers* working on *user
+   recipes* to enable them to adapt their code related to
+   *backwards-incompatible changes* (see `ESMValTool_Tutorial: issue
+   #263 <https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/263>`__).
+-  The developer or reviewer must tag the core development team to
+   notify them of the *backward-incompatible change*, and give at least
+   2 weeks for objections to be raised before merging to the main
+   branch. If a strong objection is raised the backwards-incompatible
+   change should not be merged until the objection is resolved.
+
+During the *release* process, the following information must be
+provided:
+
+-  **Release notes:** The *release* notes are already documented at
+   https://docs.esmvaltool.org/en/latest/changelog.html, and
+   “*Backwards-incompatible changes*” is the first section after
+   “Highlights”.
+
+   -  **Backwards-incompatible changes:** This section must include
+      clear instructions detailing how a *recipe developer* should adapt
+      their code for each item in this section, whether the adapted code
+      would introduce a *science change*, and the list of affected or
+      fixed *integrated recipes* that had to be updated due to the
+      *backwards-incompatible changes*, if applicable (to provide
+      further examples to *recipe developers* working on *user recipes*
+      of how to adapt code).
+   -  **Developer guidance:** *Developers* *of backwards-incompatible
+      changes* must:
+
+      -  write and include the information required for the
+         “*Backwards-incompatible changes*” section in the PR that
+         introduces the *backwards-incompatible change* [action:
```

Comment:
This action item needs to be taken care of before merging

Reply:
I added a reference to the backward compatibility policy and some related items to Checklist for pull requests" (`community/code_documentation.rst`).
In addition, I also added a reference to the policy to `doc/contributing.rst` (ESMValCore) in section "Backward compatibility": https://github.com/ESMValGroup/ESMValCore/pull/1918/. I guess that's also covered by @alistairsellar ' PR https://github.com/ESMValGroup/ESMValCore/pull/1916 (just saw this now...). Should we try to merge these to PRs?

@bouweandela, is this roughly what you had in mind?

Reply:
Thanks @axel-lauer ! I've (manually) merged ESMValGroup/ESMValTool#1916 into ESMValGroup/ESMValTool#1918 and closed the latter, so only one linked PR now.